packages/server/graphql/intranetSchema/mutations/enableSAMLForDomain.ts

```diff
@@ -19,15 +20,34 @@ const enableSAMLForDomain = {
     const r = await getRethink()
     const normalizedDomain = domain.toLowerCase()
     const normalizedUrl = url.toLowerCase()
-    await r
+
+    const samlId = await r
       .table('SAML')
-      .insert({
-        domain: normalizedDomain,
-        url: normalizedUrl,
-        metadata: metadata
-      })
+      .getAll(normalizedDomain, {index: 'domain'})
+      .nth(0)('id')
+      .default(null)
       .run()
 
+    if (samlId) {
+      await r
+        .table('SAML')
+        .get(samlId)
+        .update({
+          url: normalizedUrl,
+          metadata: metadata
+        })
+        .run()
+    } else {
+      await r
+        .table('SAML')
+        .insert({
+          domain: normalizedDomain,
+          url: normalizedUrl,
+          metadata: metadata
+        })
+        .run()
+    }
+
     return true
```

Comment:
Just wanted to point out the [difficulty of doing an atomic upsert](http://rethinkdb.1115947.n5.nabble.com/upsert-on-a-secondary-key-td983.html) in RethinkDB on a non primary key. I think it should be all right though since the saml table shouldn't be prone to race conditions.

Reply:
that's very true.
what if we start fresh & have the `id` be the domain?
then, we could do a `.insert({id: domain, metadata, url}, {conflict: 'replace'})`

Reply:
yup that's the other alternative i was thinking of! if the data needs to be cleaned anyway, which i'm assuming it is, it's probably worth it to start over again to get the guarantee and nicer API.

Reply:
Makes sense. I've pushed the update using the domain as the id

Reply:
i like it! could you write a migration that:
- clones the existing rows in the DB that changes the ID to match the domain
- delete the rows where ID !== domain

Reply:
Sure! I've just pushed an update with a new migration.

A couple of small questions/considerations I'd like to ask:
1. With my current implementation, there will be a dormant `domain` column for existing data. SAML is a relatively small table so perhaps that's not an issue? Otherwise, I could drop the column.
2. I could add a migrate down function that would clear / remove the `id` column but I'm not sure that that would be particularly useful? My understanding is that if we were renaming a column, for example, we'd want the option to migrate back to the previous name in case it caused bugs in the app. As we're simply adding data to an empty column, migrating back doesn't seem necessary to me.